code/modules/mod/modules/modules_engineering.dm

```diff
@@ -151,3 +151,100 @@
 	QDEL_NULL(chain)
 	return ..()
 
+///Atmos water tank module
+
+/obj/item/mod/module/firefighting_tank
+	name = "MOD firefighting tank"
+	desc = "A refridgerated and pressurized module tank with an extinguisher nozzle, intended to fight fires. Swaps between extinguisher, nanofrost launcher, and metal foam dispenser for breaches. Nanofrost converts plasma in the air to nitrogen, but only if it is combusting at the time."
+	icon_state = "firefighting_tank"
+	module_type = MODULE_ACTIVE
+	complexity = 2
+	active_power_cost = DEFAULT_CHARGE_DRAIN * 3
+	device = /obj/item/extinguisher/mini/mod
+
+#define EXTINGUISHER 0
+#define NANOFROST 1
+#define METAL_FOAM 2
+
+/obj/item/extinguisher/mini/mod
+	name = "modsuit extinguisher nozzle"
+	desc = "A heavy duty nozzle attached to a modsuit's internal tank."
+	icon = 'icons/obj/watertank.dmi'
+	icon_state = "atmos_nozzle_1"
+	item_state = "nozzleatmos"
+	safety = 0
+	max_water = 500
+	power = 8
+	precision = 1
+	cooling_power = 5
+	w_class = WEIGHT_CLASS_HUGE
+	flags = NODROP //Necessary to ensure that the nozzle and tank never seperate
+	var/nozzle_mode = EXTINGUISHER
+	var/metal_synthesis_charge = 0
+	COOLDOWN_DECLARE(nanofrost_cooldown)
+
+/obj/item/extinguisher/mini/mod/attack_self(mob/user)
+	switch(nozzle_mode)
+		if(EXTINGUISHER)
+			nozzle_mode = NANOFROST
+			to_chat(user, "<span class='notice'> Swapped to nanofrost launcher</span>")
+		if(NANOFROST)
+			nozzle_mode = METAL_FOAM
+			to_chat(user, "<span class='notice'> Swapped to metal foam synthesizer</span>")
+		if(METAL_FOAM)
+			nozzle_mode = EXTINGUISHER
+			to_chat(user, "<span class='notice'> Swapped to water extinguisher</span>")
+	update_icon(UPDATE_ICON_STATE)
+
+/obj/item/extinguisher/mini/mod/update_icon_state()
+	switch(nozzle_mode)
+		if(EXTINGUISHER)
+			icon_state = "atmos_nozzle_1"
+		if(NANOFROST)
+			icon_state = "atmos_nozzle_2"
+		if(METAL_FOAM)
+			icon_state = "atmos_nozzle_3"
+
+/obj/item/extinguisher/mini/mod/afterattack(atom/target, mob/user)
+	var/is_adjacent = user.Adjacent(target)
+	if(is_adjacent && AttemptRefill(target, user))
+		return
+	switch(nozzle_mode)
+		if(EXTINGUISHER)
+			..()
+			return
+		if(NANOFROST)
+			if(reagents.total_volume < 100)
+				to_chat(user, "<span class='notice'> You need at least 100 units of water to use the nanofrost launcher!</span>")
```

Comment:
generally, if something fails, you'll want to use warning instead of notice
```suggestion
				to_chat(user, "<span class='warning'> You need at least 100 units of water to use the nanofrost launcher!</span>")
```